_posts/2023-07-12-Add-client-side-HTTP-authentication-mechanism-support.adoc

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: "Upcoming client side HTTP authentication mechanisms support"
+date: 2023-07-12
+tags: http authentication client
+synopsis: The implementation of client side HTTP authentication mechanisms in Elytron is in progress.
+author: keshav-kumar
+---
+
+
+:toc: macro
+:toc-title:
+
+
+This feature will introduce a significant enhancement to WildFly Elytron - the addition of client-side HTTP authentication mechanism support. With this new feature, WildFly Elytron will provide a client-side framework for HTTP authentication mechanisms, allowing for secure and seamless communication with secured server.
+
+
+toc::[]
+
+
+== Steps for using Elytron HTTP authentication client
+Elytron HTTP client is planned to be used as follows :
+
+
+=== Configuring SSL for Secure Communication:
+All the possible configs are in this link related to SSL Configuration : https://docs.wildfly.org/28/Client_Guide.html
+
+
+Configure SSL in Wildfly Elytron: Update the Wildfly Elytron configuration file (e.g., wildfly-config.xml) to include the SSL configuration.
+
+
+=== Adding Required Credentials
+Inside Wildfly Elytron configuration file add required credential for authorizing the request
+
+
+==== Basic Authentication Mechanism and Digest Authentication Mechanism
+For Basic Authentication Mechanism, add username and password inside the configuration file
+
```

Comment:
@keshav-redhat Just a minor, you can add a sentence here explaining that appropriate headers (BASIC or DIGEST) will be added by the ElytronHttpClient to the request. The information in the headers will be obtained from elytron client configuration automatically.

Same for bearer below